File utils

AlexeyNemytov opened the pull request:
File Utils fixed for automatic generation of files - fast and dirty fix
usage example:
ruby -Ilib bin/file_utils --command=generate_files --log_write_to_console=true --log_debug_level=3

AlexeyNemytov committed:
Fast and Dirty file utils fix to work with new Params

AlexeyNemytov committed:
Fix file utils run with params

AlexeyNemytov committed:
Fix run function with new params

yarondbb commented on lib/file_utils/file_utils.rb:
```diff
@@ -11,77 +11,88 @@
+    Params.string 'cd_b', 'path' ,'b path'
+    Params.string 'cd_out', 'path' ,'output path'
+    Params.string 'cd_in', 'path' ,'input path'
+    #Params.string 'config_file', 'path' ,'configuration file path'
```
remove this line

yarondbb commented on lib/file_utils/file_utils.rb:
```diff
@@ -11,77 +11,88 @@
```
See section 1 in style guid: https://sites.google.com/site/rndbbnorth/cheat-sheet/code-styling?pli=1
 Pls addthe file author and short run description

yarondbb commented on lib/file_utils/file_utils.rb:
```diff
@@ -11,77 +11,88 @@
-      def initialize(arguments)
-        if arguments["command"] == "mksymlink"
+      def FileUtils.run
+        if Params['command'] == "mksymlink"
```
See section 13 in style guid: https://sites.google.com/site/rndbbnorth/cheat-sheet/code-styling?pli=1
pls change " to '

Do this for all cases in the files in the commit

yarondbb commented:
pls add paranthesis to all methods call (which has at lesat one parameter)

yarondbb commented on lib/file_utils/file_utils.rb:
```diff
@@ -11,77 +11,88 @@
             Log.info "--ref_cd is not set"
             return
-          end unless not arguments["ref_cd"].nil?
+          end unless not Params['ref_cd'].nil?
```
1.This structure is not human readable where the condition is found after few libes of code.
2. Also: unless not ==> becomes: if
3. error messages should be printed as errors.
4. change " to '
5. change this to all cases in file
Pls see this example:

if Params['ref_cd'].nil?
  Log.error '--ref_cd is not set'
  reurn
end

yarondbb commented on lib/file_utils/file_utils.rb:
```diff
@@ -118,42 +129,42 @@ def initialize(arguments)
 
           exist_cd = nil
-          if (arguments.has_key?"exist_cd")
+          if (Params.has_key?"exist_cd")
```
Params does not have .has_key method.
what are you trying to achive? to see if this param has value other then nil? if so you can define it as:
   Params.string('exist_cd', nil, 'exist_path')
Then ask:
   if not Params['exist_cd'].nil?

AlexeyNemytov committed:
File was modified according to code review https://github.com/kolmanv/bbfs/pull/78/files#r1263161
Tested OK on windows machine for automatic file generation function

yarondbb commented on lib/file_utils/file_utils.rb:
```diff
@@ -11,186 +15,163 @@
           cd = ContentData.new()
-          cd.from_file(arguments["cd"])
-          begin
-            Log.info "Error loading content data cd=%s" % arguments["cd"]
```
1. if it is an error we can use:
   Log.error "Error loading content data cd=%s" % arguments["cd"]
2. Need to add paranthesis to all Log methods calls and all methods in file which have one or more parameters 

pls change info to warning or error to all cases

yarondbb commented on lib/file_utils/file_utils.rb:
```diff
@@ -11,186 +15,163 @@
-            begin
-              Log.info "Error loading content data exist_cd=%s" % arguments["exist_cd"]
+            exist_cd.from_file(Params['exist_cd'])
+            if  exist_cd.nil?
```
remove resundant space
   if exist_cd.nil?

AlexeyNemytov committed:
1. added paranthesis to all Log methods calls and all methods in file which have one or more parameters

2. changed info to warning or error to all cases

3.removed resundant space line135
